### DISCUSSION TITLE
Add Bend Constraint To Softbody

### DISCUSSION THREAD

**Discussion by PiCode9560** [godot-jolt/godot-jolt#926] [Questions] [answered]
Is there a bad reason that it's not added yet?

I'm not a professional at this. But, if I understand the code correctly, it looks pretty simple to implement. I assume that it only needed to be enabled for it to work. So maybe just set the `inBendType` from [here](https://github.com/godot-jolt/godot-jolt/blob/456be6dbaca18987869180af89c17335c78acd64/src/objects/jolt_soft_body_impl_3d.cpp#L593C3-L593C95l) to either `SoftBodySharedSettings::EBendType::Dihedral` or `SoftBodySharedSettings::EBendType::Distance`. Then there's supposed to be some kind of angular stiffness variable somewhere, right?

Edit: maybe the angular stiffness have something to do with [this](https://github.com/jrouwe/JoltPhysics/blob/4084c13b676a2e7ebf9662b47dcb949179b8c4a1/Jolt/Physics/SoftBody/SoftBodySharedSettings.h#L47)

**Comment by mihe** [comment, marked as answer]
> Then there's supposed to be some kind of angular stiffness variable somewhere, right?

I have no control over the properties that are exposed on the `SoftBody3D` node, since that's entirely on the Godot side of things, which means I can't expose the bend stiffness/compliance.

It's of course possible to do something like what I did with the joints already, where I create an entirely new scene node (e.g. `JoltSoftBody3D`) that has this property on it, but I'm trying to limit how much I diverge from the standard physics interface in Godot, in order to ease any eventual transition into an engine module in Godot itself.

**Reply by PiCode9560** [reply]
Okay then, how about softbody - softbody collisions?

**Reply by mihe** [reply]
That's tracked here: jrouwe/JoltPhysics#789
